src/kubeobjects/daemonset.go

```diff
@@ -19,7 +19,7 @@ func CreateOrUpdateDaemonSet(kubernetesClient client.Client, logger logr.Logger,
 		return false, err
 	}
 
-	if !HasChanged(currentDaemonSet, desiredDaemonSet) {
+	if !HashAnnotationChanged(currentDaemonSet, desiredDaemonSet) {
```

Comment:
not sure what the intention was, but I suggest one of these
```suggestion
	if !HasAnnotationChanged(currentDaemonSet, desiredDaemonSet) {
```
```suggestion
	if !HasHashAnnotationChanged(currentDaemonSet, desiredDaemonSet) {
```